components/sup/src/manager/service/mod.rs

```diff
@@ -485,11 +487,13 @@ impl Service {
 
     fn post_run(&mut self) {
         if let Some(ref hook) = self.hooks.post_run {
-            hook.run(
+            self.post_run = hook.run(
                 &self.service_group,
                 &self.pkg,
                 self.svc_encrypted_password.as_ref(),
-            );
+            ).0 == 0;
+        } else {
+            self.post_run = true;
```

Comment:
This seems potentially confusing, since `self.post_run` doesn't just reflect whether the post-run hook succeeded, since it's `true` even if you don't have a post-run hook.

Not sure what would be a better formulation at this point; I'm really just thinking out loud here.

UPDATE: perhaps instead of keeping the result of the post-run hook around, which might be confusing since it's just really needed when we start up the service, we should instead track whether the service has been "fully initialized" (or whatever we want to call it). That way, the intention is clear, whether there's a post-run hook or not.

Reply:
I didn't go down the full initialized route - I feel like `mod.rs` and `hook.rs` need a bit of refactoring - a state machine would fix these kind of issues I think.

I want to have a go at this - but it will take some time!